File: FirmwareMS/Domain/Entities/FileSend.cs
namespace Domain.Entities;

public class FileSend : BaseEntity
{
    public string? UpdateStatus { get; set; }
    public int DeviceId { get; set; }
    public int UserId { get; set; }
    public string? File { get; set; }
    public int CurrPart { get; set; }
    public int TotParts { get; set; }
}

File: FirmwareGO/domain/entities/FileSend.go
package entities

import (
	"FirmwareGO/domain/entities/interfaces"
	_ "gorm.io/gorm"
)

type FileSend struct {
	BaseEntity           //Embedded
	UpdateStatus *string `gorm:"size:255;default:New"`
	DeviceId     int
	File         *string `gorm:"size:255"`
	CurrPart     int
	TotParts     int
}

var _ interfaces.IEntity = (*FileSend)(nil)
